src/engine/renderer/tr_shader.cpp

```diff
@@ -6711,9 +6709,18 @@ void R_ListShaders_f()
 		shaderName = shader->name;
 		shaderName += shader->defaultShader ? " (DEFAULTED)" : "";
 
-		for ( int j = 0; j < shader->numStages; j++ )
+		if ( shader->stages == shader->lastStage )
+		{
+			continue;
```

Comment:
Why?

Reply:
I added code to that condition block, printing the shader information line just before continuing, but with `-` as stage number in the list.

Previously shaders without stages were not listed. This can happen when all stages from a shader are disabled (for example a shader with a single heatHaze stage while heatHaze feature is disabled), or with specific volumetric shaders like fog.